internal/predicate/nonadminbackup_predicate.go

```diff
@@ -38,49 +39,67 @@ func getNonAdminBackupPredicateLogger(ctx context.Context, name, namespace strin
 
 // Create event filter
 func (NonAdminBackupPredicate) Create(ctx context.Context, evt event.CreateEvent) bool {
-	if _, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
-		nameSpace := evt.Object.GetNamespace()
-		name := evt.Object.GetName()
-		logger := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
-		logger.V(1).Info("Received Create NonAdminBackupPredicate")
-		return true
+	nameSpace := evt.Object.GetNamespace()
+	name := evt.Object.GetName()
+	logger := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
+	logger.V(1).Info("NonAdminBackupPredicate: Received Create event")
+	if nonAdminBackup, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
```

Comment:
is not this already checked by composite predicate?

Reply:
It is checked, but here we are using nonAdminBackup, which we need to get from evt, so it's not really check it's more of get of an object.